frontend/src/metabase/components/CollectionItemsLoader.jsx

```diff
@@ -0,0 +1,32 @@
+/* @flow */
+import React from "react";
+import EntityObjectLoader from "metabase/entities/containers/EntityObjectLoader";
+
+type Props = {
+  collectionId: number,
+  children: () => void,
+};
+
+const CollectionItemsLoader = ({ collectionId, children }: Props) => (
+  <EntityObjectLoader
+    entityType="collections"
+    entityId={collectionId}
+    children={({ object }) =>
+      object &&
+      children({
+        dashboards: object.dashboards,
+        cards: object.cards,
+        pulses: object.pulses,
+        allItems:
+          [].concat(
+            object.cards.map(c => ({ ...c, type: "card" })),
+            object.dashboards.map(d => ({ ...d, type: "dashboard" })),
+            object.pulses.map(p => ({ ...p, type: "pulse" })),
+          ) || [],
+        empty: !object.dashboards && !object.cards && !object.pulses,
```

Comment:
The code above suggests `object.dashboards` etc will never be `null` (otherwise `object.dashboards.map` will throw an error), is this supposed to be `object.dashboards.length === 0` etc?